spec/lib/common/connection-spec.js

```diff
@@ -5,113 +6,254 @@
+                            }
+                        }
+                    })
+                })
+                return this.subject.start()
+                    .catch(function (Error) {
+                        console.log(Error);
+                    });
             });
-        });
 
-        describe('connected', function () {
-            it('should be false if not connected', function () {
-                this.subject.connected.should.equal(false);
+            describe('connected', function () {
+                it('should be false if not connected', function () {
+                    this.subject.connected.should.equal(false);
+                });
+
+                it('should be true if connected', function () {
+                    var self = this;
+                    amqp.createConnection = sinon.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                callback(null, {});
+                            }
+                        })
+                    })
+                    return this.subject.start().then(function () {
+                        return self.subject.connected.should.equal(true);
+                    });
+                });
             });
 
-            it('should be true if connected', function () {
-                var self = this;
+            describe('exchanges', function () {
+                it('should be undefined if not connected', function () {
+                    expect(this.subject.exchanges).equal(undefined);
+                });
 
-                return this.subject.start().then(function () {
-                    return self.subject.connected.should.equal(true);
+                it('should have exchanges if connected', function () {
+                    var self = this;
+
+                    amqp.createConnection = sinon.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                callback(null, {});
+                            },
+                            'exchanges':
+                                'on.test'
+                        })
+                    })
+                    this.subject.start().then(function () {
+
+                        self.subject.exchanges.should.not.be.undefined;
+                    });
                 });
             });
-        });
 
-        describe('exchanges', function () {
-            it('should be undefined if not connected', function () {
-                expect(this.subject.exchanges).equal(undefined);
-            });
+            describe('exchange', function ()  {
+                it('should not provide an exchange if connection not established ', function () {
+                    var self = this;
+                    return this.subject.exchange()
+                        .catch(function (Error) {
+                            console.log(Error);
```

Comment:
Same as above, I don't think we want to catch here because it will prevent the test from failing.